### PR TITLE
Fix split_strings

### PR DESCRIPTION
**Pull request by zurk**
For some reason len function for numpy array gives weird error sometimes: 
```
    if len(lengths) == 0 and len(strings) == 0:
TypeError: 'int' object is not callable
```

lengths.shape[0] works smoothly.

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
@zurk I need a test extension which actually checks this, because your previous test passed.

**Comment by zurk** [comment]
That's why It is weird and usually, it is ok to get the length of numpy array.
```python
import numpy
len(numpy.array([]))
Out[3]: 0
len(numpy.array([1,2,3]))
Out[4]: 3
len(numpy.array(["1","2","34"]))
Out[5]: 3
``` 

I will try to google the error I had.

**Comment by zurk** [comment]
it is an asdf issue. if you save and load `numpy.ndarray` it has `int` in its `__len__` magic method.
So I cannot add tests because it is a bug in asdf. Going to report the issue.

**Comment by zurk** [comment]
@vmarkovtsev here https://github.com/spacetelescope/asdf/issues/671
let's merge the PR, I cannot add a test as you asked, because actually, it should work.

**Comment by vmarkovtsev** [comment]
Go ahead and add the ASDF test - exactly how you wrote in the issue. It is a feature, not a bug.

File: modelforge/tests/test_model.py [hunk 1]
```diff
@@ -445,13 +445,17 @@ def test_save_create_missing_dirs(self):
...
-    def test_empty_split_and_merge(self):
+    def test_empty_split_save_load_merge(self):
         strings = []
         merged = merge_strings(strings)
         assert_array_equal(merged["strings"], numpy.array([], dtype="S1"))
         assert_array_equal(merged["lengths"], numpy.array([], dtype=int))
         self.assertIsNone(merged["str"])
-        strings_restored = split_strings(merged)
+        af = asdf.AsdfFile(merged)
+        with tempfile.NamedTemporaryFile(prefix="modelforge-test-", suffix=".asdf") as f:
```
Comment (vmarkovtsev) [inline modelforge/tests/test_model.py]: We can use `io.BytesIO` and avoid creating a real file.

Comment (zurk) [reply modelforge/tests/test_model.py]: fixed

**Pull request merged**